.gitlab-ci.yml

```diff
@@ -63,18 +38,35 @@ before_script:
   CI_PLATFORM: "do"
   SSH_USER: root
 
+.packet_variables: &packet_variables
+  CI_PLATFORM: "packet"
+  SSH_USER: kubespray
 
-.testcases: &testcases
-  <<: *job
-  <<: *docker_service
+.packet-job: &packet-job
+  tags:
+    - packet
+  image: quay.io/kubespray/kubespray:v2.8.1
+
+.gce-job: &gce-job
+  tags:
+    - kubernetes
+    - docker
+  image: quay.io/kubespray/kubespray:v2.8.1
+
+.docker_service: &docker_service
+  services:
+     - docker:dind
+
+.cache: &cache
   cache:
     key: "$CI_BUILD_REF_NAME"
     paths:
       - downloads/
       - $HOME/.cache
+
+.testcases: &testcases
   before_script:
-    - docker info
-    - /usr/bin/python -m pip install -r requirements.txt
+    #- /usr/bin/python -m pip install -r requirements.txt
```

Comment:
same it's used:

Logs from a job
```
/usr/bin/python -m pip install -r requirements.txt[0;m
Requirement already satisfied: ansible!=2.7.0,>=2.5.0 in /usr/local/lib/python2.7/dist-packages (from -r requirements.txt (line 1)) (2.7.4)
Requirement already satisfied: jinja2>=2.9.6 in /usr/local/lib/python2.7/dist-packages (from -r requirements.txt (line 2)) (2.10)
Requirement already satisfied: netaddr in /usr/local/lib/python2.7/dist-packages (from -r requirements.txt (line 3)) (0.7.19)
Requirement already satisfied: pbr>=1.6 in /usr/local/lib/python2.7/dist-packages (from -r requirements.txt (line 4)) (5.1.1)
```

Reply:
It's run later in the testcases part.

Reply:
where? I don't find the line

Reply:
`tests/requirements.txt` contains:

`-r ../requirements.txt`

Reply:
thanks